pokemon-league-participation-manager/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/entities/Badge.java

```diff
@@ -0,0 +1,79 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.entities;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+import java.util.Objects;
+
+/**
+ * Entity class for badge object.
+ *
+ * @author Michal Mokros 456442
+ */
+@Entity(name = "Badge")
+@Table(name = "TABLE")
+public class Badge {
+
+    @Id
+    @GeneratedValue(strategy =  GenerationType.IDENTITY)
+    private Long id;
+
+    @OneToMany
+    @NotNull
+    private Trainer trainer;
+
+    @OneToMany
```

Comment:
Same as trainer.

Reply:
So you want this to be @ManyToMany then? Because seeing the diagrams document one badge can belong to only one trainer, and trainer can have multiple badges so @ManyToOne seems okay to me.

Reply:
AFAIK @ OneToMany is on the side of the one entity that has multiple association to a different entity. What I mean is that here it should be the opposite annotation @ ManyToOne, since we are on the side of the entity where each instance has exactly one association with the other entity.